drivers/bridge/bridge.go

```diff
@@ -959,6 +961,9 @@ func (d *driver) CreateEndpoint(nid, eid string, ifInfo driverapi.InterfaceInfo,
 	if endpoint.addrv6 == nil && config.EnableIPv6 {
 		var ip6 net.IP
 		network := n.bridge.bridgeIPv6
+		if config.AddressIPv6 != nil {
+			network = config.AddressIPv6
+		}
 		ones, _ := network.Mask.Size()
 		if ones <= 80 {
 			ip6 = make(net.IP, len(network.IP))
```

Comment:
Thanks for catching this.
It cannot happen in the docker code, but this condition needs nevertheless to be covered in the libnetwork context.
Will update the diffs shortly.